whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java

```diff
@@ -466,21 +467,31 @@ private static List<Entity> createContactEntities(final RpslObject rpslObject) {
         for (final Map.Entry<CIString, Set<AttributeType>> entry : contacts.entrySet()) {
             final Entity entity = new Entity();
             entity.setHandle(entry.getKey().toString());
+            final Set<ObjectType> objectPossibleTypes = new HashSet<>();
             for (final AttributeType attributeType : entry.getValue()) {
+                objectPossibleTypes.addAll(attributeType.getReferences());
```

Comment:
Move this logic into a separate method? e.g. `@Nullable RpslObject lookup(final ObjectType objectType, final CIString primaryKey)`

Is is possible that there can ever be more than 1 reference?

Reply:
Done the refactor